notify/rocketchat/rocketchat.go

```diff
@@ -0,0 +1,266 @@
+type PostMessage struct {
+	Channel     string                              `json:"channel,omitempty"`
+	Text        string                              `json:"text,omitempty"`
+	ParseUrls   bool                                `json:"parseUrls,omitempty"`
+	Alias       string                              `json:"alias,omitempty"`
+	Emoji       string                              `json:"emoji,omitempty"`
+	Avatar      string                              `json:"avatar,omitempty"`
+	Attachments []Attachment                        `json:"attachments,omitempty"`
+	Actions     []config.RocketchatAttachmentAction `json:"actions,omitempty"`
+}
+
+// New returns a new Rocketchat notification handler.
+func New(c *config.RocketchatConfig, t *template.Template, l log.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	client, err := commoncfg.NewClientFromConfig(*c.HTTPConfig, "rocketchat", httpOpts...)
+	if err != nil {
+		return nil, err
+	}
+	token, err := getToken(c)
+	if err != nil {
+		return nil, err
+	}
+	tokenID, err := getTokenID(c)
+	if err != nil {
+		return nil, err
+	}
+	return &Notifier{
+		conf:         c,
+		tmpl:         t,
+		logger:       l,
+		client:       client,
+		retrier:      &notify.Retrier{},
+		postJSONFunc: notify.PostJSON,
+		token:        token,
+		tokenID:      tokenID,
+	}, nil
+}
+
+func getTokenID(c *config.RocketchatConfig) (string, error) {
+	if len(c.TokenIDFile) > 0 {
+		content, err := os.ReadFile(c.TokenIDFile)
+		if err != nil {
+			return "", fmt.Errorf("could not read %s: %w", c.TokenIDFile, err)
+		}
+		return strings.TrimSpace(string(content)), nil
+	}
+	return string(*c.TokenID), nil
+}
+
+func getToken(c *config.RocketchatConfig) (string, error) {
+	if len(c.TokenFile) > 0 {
+		content, err := os.ReadFile(c.TokenFile)
+		if err != nil {
+			return "", fmt.Errorf("could not read %s: %w", c.TokenFile, err)
+		}
+		return strings.TrimSpace(string(content)), nil
+	}
+	return string(*c.Token), nil
+}
+
+// Notify implements the Notifier interface.
+func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	var err error
+
+	data := notify.GetTemplateData(ctx, n.tmpl, as, n.logger)
+	tmplText := notify.TmplText(n.tmpl, data, &err)
+	if err != nil {
+		return false, err
+	}
+	titlex := tmplText(n.conf.Title)
+	if err != nil {
+		return false, err
+	}
+
+	title, truncated := notify.TruncateInRunes(titlex, maxTitleLenRunes)
+	if truncated {
+		key, err := notify.ExtractGroupKey(ctx)
+		if err != nil {
+			return false, err
+		}
+		level.Warn(n.logger).Log("msg", "Truncated title", "key", key, "max_runes", maxTitleLenRunes)
+	}
+	att := &Attachment{
+		Title:     title,
+		TitleLink: tmplText(n.conf.TitleLink),
+		Text:      tmplText(n.conf.Text),
+		ImageURL:  tmplText(n.conf.ImageURL),
+		ThumbURL:  tmplText(n.conf.ThumbURL),
+		Color:     tmplText(n.conf.Color),
+	}
+	numFields := len(n.conf.Fields)
+	if numFields > 0 {
```

Comment:
Is there a limit on the maximum number of fields that we need to enforce?

Reply:
I don't think so. This is basically copy and paste from the slack notifier, which we are using now see (https://github.com/prometheus/alertmanager/issues/3546) and it handled 100+ alerts in a group just fine.
I haven't experienced any limits so far, also looking at the [RocketChat SDK](https://github.com/RocketChat/Rocket.Chat.Go.SDK/tree/master/rest) I don't see any limitations here.

Reply:
Ah OK! Sounds fine to me. If needed we can add them later.